multimodal/src/autogluon/multimodal/utils/metric.py

```diff
@@ -76,70 +79,93 @@ def infer_metrics(
+                validation_metric_name = eval_metric_name
+                return validation_metric_name, eval_metric_name
+            elif eval_metric_name.lower() in RETRIEVAL_METRICS:
+                # Currently only support recall as validation metric in retrieval tasks.
+                validation_metric_name = RECALL
+                return validation_metric_name, eval_metric_name
+            else:
+                if not is_customized:
+                    warnings.warn(
+                        f"Metric {eval_metric_name} is not supported as the evaluation metric for {problem_type} in matching tasks."
+                    )
+
+        # Fallback based on problem type unless it's a customized metric
+        if problem_type is None:
+            validation_metric_name, fallback_eval_metric_name = MATCHING_METRICS_WITHOUT_PROBLEM_TYPE
+            logger.info(
+                f"Metric {eval_metric_name} and metric {validation_metric_name} are used "
+                f"as the evaluation metric and the validation metric for matching tasks by default. "
+            )
+        elif problem_type in MATCHING_METRICS:
+            validation_metric_name, fallback_eval_metric_name = MATCHING_METRICS[problem_type]
+            logger.info(
+                f"Metric {eval_metric_name} and metric {validation_metric_name} are used "
+                f"as the evaluation metric and the validation metric for {problem_type} in matching tasks by default. "
+            )
+        else:
+            raise NotImplementedError(f"Problem type: {problem_type} is not yet supported for matching!")
+        if not is_customized:
+            eval_metric_name = fallback_eval_metric_name
+        logger.info(
+            f"Metric {eval_metric_name} and metric {validation_metric_name} are used "
+            f"as the evaluation metric and the validation metric for {problem_type} in matching tasks by default. "
+        )
+        return validation_metric_name, eval_metric_name
 
     if eval_metric_name is not None:
-        if problem_type != BINARY and eval_metric_name.lower() in [
-            ROC_AUC,
-            AVERAGE_PRECISION,
-            F1,
-        ]:
-            raise ValueError(f"Metric {eval_metric_name} is only supported for binary classification.")
-
-        if eval_metric_name in VALID_METRICS:
+        # Infer evaluation metric
+        if eval_metric_name.lower() not in problem_property.supported_evaluation_metrics and not is_customized:
+            warnings.warn(
+                f"Metric {eval_metric_name} is not supported as the evaluation metric for {problem_type}. "
+                f"The evaluation metric is changed to {problem_property.fallback_evaluation_metric} by default."
+            )
+            if problem_property.fallback_evaluation_metric is not None:
+                eval_metric_name = problem_property.fallback_evaluation_metric
+            else:
+                return None, None
```

Comment:
Instead of adding the `return None, None`, how just remove `if problem_property.fallback_evaluation_metric is not None:`?

Reply:
but then we still need to check if the eval_metric_name is None while inferring validation metric below